### DISCUSSION TITLE
Nested bullet points are flattened

### DISCUSSION THREAD

**Discussion by YoeriVD** [structurizr/lite#36] [Q&A]
Hi Simon

For a (short-lived) project we're experimenting going all-in structurizr for diagrams and documentation, as the workflow is very appealing.

I know you have stated that not all markdown features might be supported:

> [markdown-it](https://markdown-it.github.io/) is used to render Markdown, and [asciidoctor.js](https://github.com/asciidoctor/asciidoctor.js) is used to render AsciiDoc. Please note that Structurizr's documentation feature may not render your Markdown/AsciiDoc in exactly the same way that other dedicated documentation tools will, and some features may not be supported. If you need full control over the look and feel of your documentation, you may be better to use an external tool (e.g. Asciidoctor, Hugo, Jekyll, etc) in conjunction with the [iframe embed](https://structurizr.com/help/embed-iframe) or [image embed](https://structurizr.com/help/embed-image) features.

But I think this (nested lists) still counts as an elemental feature in order to write even the most simple forms of documentation.

![image](https://user-images.githubusercontent.com/2865793/219941435-3a076835-90ed-4a37-a8bf-f0a8acb49f22.png)

Any chance this is fixable? Or do you have a workaround?

**Comment by simonbrowndotje** [comment]
Does this work in markdown-it?

**Reply by YoeriVD** [reply]
it does

![image](https://user-images.githubusercontent.com/2865793/219954859-62a45600-9244-4a17-af43-c2e47a717d3d.png)

**Comment by simonbrowndotje** [comment]
2 or more spaces seems to work for me.

<img width="257" alt="Screenshot 2023-02-19 at 13 58 32" src="https://user-images.githubusercontent.com/1009874/219952977-27224513-bfa2-4046-bdf4-2ab860a45ba5.png">
<img width="746" alt="Screenshot 2023-02-19 at 13 58 58" src="https://user-images.githubusercontent.com/1009874/219952979-b175dbff-646e-4541-bc84-71e0e2f210e6.png">

**Comment by YoeriVD** [comment]
@simonbrowndotje  It works in every markdown renderer I try, including markdown-it and vs code preview.
But unfortunately when rendered in structurizr lite, the hierarchy is gone. I think I have tried any amount of spaces so far.

![image](https://user-images.githubusercontent.com/2865793/219954724-1666c1d6-39d9-4a33-81f5-235b5af2749f.png)

**Comment by YoeriVD** [comment]
@simonbrowndotje 
Tried it in the on prem version as well, with the same result


![image](https://user-images.githubusercontent.com/2865793/221660890-43933bd7-4aeb-4a89-ba5f-3431a05f9989.png)

**Comment by mikeleo** [comment]
@simonbrowndotje

An issue appears to be in at least plugins PlantUMLEncoderPlugin and MermaidEncoderPlugin if the plugins/plugins.jar is in the workspace directory. 

While trying to find embedded content, every line is trimmed and then content is updated with the trimmed lines, breaking markdown where the whitespace is significant. 

https://github.com/structurizr/examples/blob/e7ca0b03e32ffb663221470ef9a359362716bca8/dsl/plantuml-and-mermaid/plugin/src/main/java/plantuml/PlantUMLEncoderPlugin.java#L51

A trimmed line should only be used when looking for the embedded PlantUML or Mermaid content. If it falls through to pushing the line back onto the content, then it should use original, untrimmed line.
